api/app/controllers/user.js

```diff
@@ -116,24 +155,100 @@ module.exports = {
+            );
+            if (!user) {
+                return res.status(404).json({
+                    message: "User not found",
+                });
+            }
+            userUpdate = {
+                ...userUpdate,
+                ...req.body
+            }
+            userUpdate.save();
+            res.json(userUpdate);
+
+        } catch (error) {
+            return res.status(400).send({
+                message: "Internal server error. Please retry later",
+            });
+        }
     },
-    updatePassword(req, res) {
+    async updatePassword(req, res) {
         // TODO: Update the user password in the database. Ask for the old password, verify it, then re-insert the new one after checking its strength and hashing it
-        res.json("OK");
+
+        try {
+            const userUpdatePassword = await User.findByPk(
+                req.user.id
+            );
+            if (!user) {
+                return res.status(404).json({
+                    message: "User not found",
+                });
+            }
+            user = {
+            ...userUpdatePassword.password
+            }
+            userUpdatePassword.save();
+            res.json(userUpdatePassword);
+
+            const isMatch = await bcrypt.compare(password, user.password);
+
+            if (!isMatch) {
+                return res.status(401).json({
+                    message: "Incorrect password",
+                    password,
+                });
+            }
+
+        } catch (error) {
+            return res.status(400).send({
+                message: "Internal server error. Please retry later",
+            });
+        }
     },
-    deleteAccount(req, res) {
+    async deleteAccount(req, res) {
         // TODO: Delete the user account. Request the current password, and change the account status to inactive.
+            try {
+                userToDelete = User.findByPk(
+                    req.user.id
+                );
+                if (!user) {
+                    return res.status(404).json({
+                        message: "User not found",
+                    });
+                };
+                const isMatch = await bcrypt.compare(password, user.password);
+
+            if (!isMatch) {
+                return res.status(401).json({
+                    message: "Incorrect password",
+                    password,
+                });
+            }
+
+            } catch (error) {
+                return res.status(400).send({
+                    message: "Internal server error. Please retry later",
+                }); 
+            }
         // TODO: (Add a database migration, add a column "account_is_deactivated" and "date_of_deactivation" ). After 15 days, if the user has not reconnected, delete the account and the related data. So you have to add a function to set account_is_deactivated back to false in the login method
+        res.json("Currently not implemented");
```

Comment:
Delete this line

Reply:
no , is not the good commit look at the latest ::

Reply:
Sry i look the first commit